src/do_command.c

```diff
@@ -581,6 +581,14 @@ static int child_process(entry * e, char **jobenv) {
 			Debug(DPROC,
 				("[%ld] no more grandchildren--mail written?\n",
 					(long) getpid()));
+
```

Comment:
You should have put it before line 592 (after the for cycle) and not here inside it.

Reply:
Ah, yes. A bit quick in removing the old code, and not reviewing what I left. Sorry about that.
I have now submitted a new commit.